lib/mayaUsd/commands/layerEditorCommand.cpp

```diff
@@ -868,7 +883,7 @@ class LockLayer : public BaseCmd
         // Execute lock commands
         for (size_t layerIndex = 0; layerIndex < _layers.size(); layerIndex++) {
             // Note: the undo of system-locked is unlocked by design.
-            if (_previousStates[layerIndex] == MayaUsd::LayerLockType::LayerLock_SystemLocked) {
+            if (_lockType == MayaUsd::LayerLockType::LayerLock_SystemLocked) {
```

Comment:
I believe this should also be conditioned?

Reply:
I don't understand what you mean

Reply:
The previous state will always have the correct value when undoing, so there is no need to protect, at worst it will set the same state it already was.

Reply:
As long as any consecutive number of system-locks results in Unlock, then it's all good.